js/application.js

```diff
@@ -0,0 +1,356 @@
+    monsterCol = col;
+    log("monster position" + monsterCol);
+    
+}; 
+
+// set branch head (head --> end)
+var setBranch = function(row, col){
+    if((row<game.frame.length)){
+        game.frame[row][col] = branch;
+        game.frame[row][col+1] = branch;
+    }
+    branchPos = [row,col]; 
+};
+
+//clear position and clear screen
+var clearPosition = function(row, col){
+    if(row<frame.length-1){
+    frame[row][col] = 0;
+    }
+};
+
+var clearScreen = function(row, col){
+    for(row=0; row < frame.length; row++){
+        for(col =0; col < colNum; col++){
+            frame[row][col] = 0;
+        }
+    }
+};
+
+//timer for Set up and respond
+var minusSecond = function(){
+  timeLeft --;
+  return timeLeft;
+};
+
+var countDown = setInterval(function(){
+    minusSecond();
+    if(timeLeft <= 0){clearInterval(countDown)}
+},1000)
+
+//test set up
+var test = function(){
+    setMonster(3);
+    setBranch(1,2);
+    setBranch(1,3);
+    setBranch(3,2);
+    setBranch(3,3);
+    setBranch(6,2);
+    setBranch(6,3);
+    setPig(2);
+    log(game.frame);
+};
+
+// movement
+var checkRightSpot = function(){
+    if((monsterCol<colNum-1)&&(game.frame[monsterRow][monsterCol+1] !== 0)&&(game.frame[monsterRow][monsterCol+1] !== 9)){
+    game.frame[monsterRow][monsterCol+1] = monster;
+    monsterCol++;
+    log(game.frame);
+    log('right');
+    return true
+} 
+ return false   
+};
+
+var checkLeftSpot = function(){
+    if((monsterCol-1>=0)&&(game.frame[monsterRow][monsterCol-1] !== 0)&&(game.frame[monsterRow][monsterCol-1] !== 9)){
+    game.frame[monsterRow][monsterCol-1] = monster;
+    monsterCol--;
+    log(game.frame)
+    log('left')
+    return true;
+} 
+    return false;
+};
+    
+var checkDownSpot = function(){
+    if(monsterRow<game.frame.length){
+    monsterRow++;
+    game.frame[monsterRow][monsterCol] = monster;
+    log(game.frame)
+    log('down')
+    return true
+} 
+    return false
+}   ; 
+
+var move = function(){
+        if(checkRightSpot() === true){
+            checkRightSpot();
+        } else if (checkLeftSpot() === true){
+            checkLeftSpot();
+        } else {
+            checkDownSpot();
+        }
+};
+
+var winner = function(){
+    if(monsterCol === pigCol){
+        console.log ("Bloody pig is gone");
+    } else {
+        console.log ("Pig is still alive! Monster is dumb dumb")
+    }
+    
+}
+
+
+
+$(document).ready(function(){
+// colNum = parseInt($('.numOfColumn').val());
+// colWidth = 700/(colNum);
+
+
+
+$(document).on('click', '.indiBranch', function(){
+    console.log("indiBranch");
+    $(this).css('stroke', 'rgba(139,69,19,1)')
+
+});
+
+$(document).on('dblclick', '.indiBranch', function(){
+    console.log("indiBranch");
+    $(this).css('stroke', 'rgba(139,69,19,0)')
+
+});
+
+
+        
+//draw background
+    $('.go').click(function(){
+      console.log("go button clicked")
+      
+
+    })
+
+ 
+    
+
+
+
+
+
+
+
+
+
+
+
+
+})//end of jQuery
+
```

Comment:
Delete from here bellow, if you want to keep the code, put it in a different branch ;)